testing/citest/tests/gcs_pubsub_gae_test.py

```diff
@@ -0,0 +1,453 @@
+    path = os.path.join('pipelines', self.TEST_APP, pipeline_id)
+
+    builder = st.HttpContractBuilder(self.agent)
+    (builder.new_clause_builder('Has Pipeline',
+                                retryable_for_secs=5)
+     .get_url_path(
+       'applications/{app}/pipelineConfigs'.format(app=self.TEST_APP))
+     .excludes_path_value('name', pipeline_id))
+
+    return st.OperationContract(
+      self.new_delete_operation(
+        title='delete_bake_pipeline', data=payload, path=path,
+        status_class=st.SynchronousHttpOperationStatus),
+      contract=builder.build())
+
+class GcsPubsubGaeTest(st.AgentTestCase):
+  @classmethod
+  def setUpClass(cls):
+    runner = citest.base.TestRunner.global_runner()
+    scenario = runner.get_shared_data(GcsPubsubGaeTestScenario)
+    bindings = scenario.bindings
+
+    branch = bindings['BRANCH']
+    git_repo = bindings['GIT_REPO_URL']
+
+    gcs_path = 'gs://{bucket}'.format(bucket=scenario.bucket)
+    topic = '{}-topic'.format(bindings['TEST_APP'])
+    subscription = '{}-subscription'.format(bindings['TEST_APP'])
+
+    # App to tar and upload to GCS.
+    command = 'git clone {repo} -b {branch} {dir}'.format(
+      repo=git_repo, branch=branch, dir=scenario.temp)
+    logging.info('Fetching %s', git_repo)
+    subprocess.Popen(command, stderr=sys.stderr, shell=True).wait()
+
+
+  @classmethod
+  def tearDownClass(cls):
+    runner = citest.base.TestRunner.global_runner()
+    scenario = runner.get_shared_data(GcsPubsubGaeTestScenario)
+    bindings = scenario.bindings
+    shutil.rmtree(scenario.temp)
+
+  @property
+  def scenario(self):
+    return citest.base.TestRunner.global_runner().get_shared_data(GcsPubsubGaeTestScenario)
+
+  @property
+  def testing_agent(self):
+    return self.scenario.agent
+
+  def test_a_create_app(self):
+    self.run_test_case(self.scenario.create_app(),
+                       retry_interval_secs=8, max_retries=8)
+
+  def test_b_create_pipeline(self):
+    self.run_test_case(self.scenario.create_deploy_pipeline(),
+                       retry_interval_secs=8, max_retries=8)
+
+  def test_d_run_pipeline(self):
+    time.sleep(60)
+    # Wait for Echo's cache to pick up the deploy pipeline.
+    # This is generally a bad strategy for synchronizing cache timing, so we'll investigate
+    # performing this check a different way in the future. One option is querying the
+    # Spinnaker services' metrics endpoint and inspecting metrics related to caching
+    # once they are instrumented.
+    self.run_test_case(self.scenario.trigger_deploy_pipeline(), poll_every_secs=5)
+
+  def test_x_delete_load_balancer(self):
+    self.run_test_case(self.scenario.delete_load_balancer(),
+                       retry_interval_secs=8, max_retries=8)
+
+  def test_y_delete_pipeline(self):
+    self.run_test_case(
+      self.scenario.delete_deploy_pipeline(self.scenario.pipeline_id))
+
+  def test_z_delete_app(self):
+    time.sleep(60) # Wait for cache update.
```

Comment:
we never do this in any other test and they all delete apps except for the kato tests. What's different here?

Reply:
Removed.